authors/apps/article/tests/test_like_an_article.py

```diff
@@ -0,0 +1,149 @@
+from django.urls import reverse
+import os
+from ...authentication.models import User
+from ..models import Article
+from rest_framework.test import APIClient
+from rest_framework.test import APITestCase
+from rest_framework import status
+
+
+class TestLikeArticle(APITestCase):
+    """ class for testing email verification"""
+
+    def setUp(self):
+        """
+        Prepare test environment for each testcase
+        """
+
+        self.client = APIClient()
+        self.article = Article()
+        self.signup_url = reverse('authentication:register')
+        self.user_details = {
+            'user': {
+                'username': 'user1',
+                'email': 'evajohnson714@gmail.com',
+                'password': 'somepass12345',
+            }
+        }
+        resp = self.client.post(
+            self.signup_url,
+            self.user_details,
+            format='json')
+        self.token = resp.data['token']
+        self.email = "test_user@gmail.com"
+        self.name = "test"
+        self.user = User(username=self.name, email=self.email)
+        self.user.set_password("@Winners11")
+        self.user.save()
+        self.user_id = User.objects.get(email=self.email).pk
+        self.slug = "this-is-a-question"
+        self.title = "this is a question"
+        self.description = "this is a description"
+        self.body = "this is a body"
+        author = self.user
+        article = Article(
+            user=author,
+            slug=self.slug,
+            body=self.body,
+            title=self.title,
+            description=self.description + " hhj"
+        )
+        article.save()
+        self.rate_details = {
+            "user": {
+                "slug": self.slug,
+                "rate": 3
+            }
+        }
+        self.data = {}
+        self.like_url = os.environ["URL"] + \
+            "api/article/" + "like/" + self.slug + "/"
+        self.view_likes_url = os.environ["URL"] + "api/article/rate/"
+        self.articles_url = os.environ["URL"] + "api/article/"
+        self.create_articles_url = os.environ["URL"] + "api/article/create"
```

Comment:
For modularity, instead of repeating the setup for every test. Why dont you create a base setup file so that they could share the setup files. It will also help anyone else writing tests could just add up to that.